code-challenges.test.js

```diff
@@ -1,56 +1,139 @@
+
+// // a) Create a test with expects statement using the variable provided.
+
+describe("singleLetter",() => {
+
+    it("takes in an array of words and a single letter and returns all the words that contain that particular letter.", () => {
+       
+        const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+        const letterA = "a"
+        // Expected output: ["Apple", "Banana", "Orange"]
+        const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+        const letterE = "e"
+        // Expected output: ["Cherry", "Blueberry", "Peach"]
+
+      expect(singleLetter(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+      expect(singleLetter(arrayOfWords2, letterE )).toEqual(["Cherry", "Blueberry", "Peach"])
+        })
+    })
 
 // b) Create the function that makes the test pass.
-
-
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+//pseudocode
+//takes in an array of words and a single letter and returns all the words that contain that particular letter.
+//create a function called "singleLetter" that takes in an array and a letter
+//create a variable called "bigA" to pass the parameter letter to be also uppercase
+// create a variable called "returnWord" that will filter in the array and will return if the value is true using the method .includes?
+//return value
+
+const singleLetter = (array, letter) => {
+    let bigA = letter.toUpperCase()
+    let returnWord = array.filter(value => {
+    if (value.includes(letter) === true || value.includes(bigA) === true ) {
+        return value
+    }
+    })
+    return returnWord
+    }
+
+
+// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
+describe("isItFullHouse",() => {
+
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house.", () => {
 
+       const hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+       const hand2 = [5, 5, 3, 3, 4]
+       // Expected output: false
+      const hand3 = [5, 5, 5, 5, 4]
+      // Expected output: false
+       
+      expect(isItFullHouse(hand1)).toEqual(true)
+      expect(singleLetter(hand2)).toEqual(false)
+      expect(singleLetter(hand3)).toEqual(false)
+        })
+    })
 
 // b) Create the function that makes the test pass.
+//pseudocode 
+//create a function "isItFullHouse"
+
+
+//Sorry, I tried to do this one but I was not able to quite understand the question.  :/
```

Comment:
Did you reach out for help or for clarity?  Your code seems to be heading in the right direction however I don't see any pseudocode for what you have written.  When stuck, a great way to see what the intent of the challenge is would be to see what the expected output is.  If you are stuck after that, research, ask an instructor, or ask your classmates.